### PR TITLE
[Backport release-21.05] betterlockscreen 3.2.0 -> 4.0.0

### PR DESCRIPTION
**Pull request by SebTM**
(cherry picked from commit 2f59b54e3f1318c4e0a6079baebf621b9b402880)
**Please note changes from 3.2 to 4.0 are breaking, depending on how/how intense Betterlockscreen is used. But as there are multiple fundamental bugs in v3.2 we are not able to provide fixes for this.** 

###### Motivation for this change
Backport new major release after taking-over maintenance for the project

###### Things done
Adjusted derivation and included all dependencies

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/screensavers/betterlockscreen/default.nix [hunk 1]
```diff
@@ -20,18 +34,22 @@ stdenv.mkDerivation rec {
...
-        [imagemagick i3lock-color xdpyinfo xrandr bc feh procps xrdb];
+        [ bc coreutils i3lock-color gawk gnugrep gnused imagemagick procps xdpyinfo xrandr xset ];
     in ''
+      runHook preInstall
+
       mkdir -p $out/bin
       cp betterlockscreen $out/bin/betterlockscreen
       wrapProgram "$out/bin/betterlockscreen" --prefix PATH : "$out/bin:${PATH}"
+
+      runHook preInstall
```
Comment (hrhino) [inline pkgs/misc/screensavers/betterlockscreen/default.nix:45]:
```suggestion
      runHook postInstall
```

**Review by hrhino — COMMENTED** [review]
Please split this into two commits: "add sebtm as maintainer" and "betterlockscreen: 3.2.0 -> 4.0.0"

**Comment by hrhino** [comment]
Also, this doesn't look like it works if i have `i3lock` also installed: here's a relevant bit from `bin/betterlockscreen`:
```sh
    if ! cmd_exists "i3lockcolor_bin" && cmd_exists "i3lock"; then
        i3lockcolor_bin="i3lock"
    fi
```
where I think it means to say `cmd_exists "$i3lockcolor_bin`... currently it's going to replace that with `i3lock` since no package provides `i3lockcolor_bin`.

**Comment by SebTM** [comment]
Thanks for your Feedback, split the commits and fixed the wrong hook. Please also note #133417 to fix it in master which got already merged.

**Comment by SebTM** [comment]
> Also, this doesn't look like it works if i have `i3lock` also installed: here's a relevant bit from `bin/betterlockscreen`:
> 
> ```shell
>     if ! cmd_exists "i3lockcolor_bin" && cmd_exists "i3lock"; then
>         i3lockcolor_bin="i3lock"
>     fi
> ```
> 
> where I think it means to say `cmd_exists "$i3lockcolor_bin`... currently it's going to replace that with `i3lock` since no package provides `i3lockcolor_bin`.

It's really wanted that way because Arch Linux populates i3lock-color as i3lock and this caused a high support come up...

**Comment by hrhino** [comment]
> Thanks for your Feedback, split the commits and fixed the wrong hook. Please also note #133417 to fix it in master which got already merged.

Gah! I missed that this was a backport. Ah well; sorry. I think the hook change should also be backported? I'm not a release manager; probably someone with merge bits will have to opine here.

> It's really wanted that way because Arch Linux populates i3lock-color as i3lock and this caused a high support come up...

Hm fair; I think you just need to add that `$` to `i3lockcolor_bin` so it checks for the right thing (`i3lock-color` not `i3lockcolor_bin`)

**Comment by SebTM** [comment]
Oh, I didn't got that thanks for pointing out!

**Comment by Artturin** [comment]
since it has breaking changes lets not backport it

**Pull request closed without merging**